src/plugins/share/public/components/tabs/export/export_content.tsx

```diff
@@ -168,42 +173,56 @@ const ExportContentUi = ({
     return (
       <EuiButton
         fill
-        color={isDirty ? 'warning' : 'primary'}
+        color={(isDirty && renderCopyURLButton) || warnings.length > 0 ? 'warning' : 'primary'}
         onClick={getReport}
         data-test-subj="generateReportButton"
         isLoading={isCreatingExport}
       >
-        {generateExportButton}
+        {generateExportButton ?? (
```

Comment:
Any idea why the button shows `Export file` instead of `Download CSV` for the button in Lens when the CSV radio is selected? It looks like it should be getting overridden here:

https://github.com/nickofthyme/kibana/blob/78ddf771b75666287abb3272021e0c6507c0fb87/x-pack/plugins/lens/public/app_plugin/csv_download_provider/csv_download_provider.tsx#L168 

BTW, this isn't working in main, either. But the original intention was to convey that the CSV file will be instantly downloaded, as opposed to PNG/PDF which kicks of a report job.